File: source/update/dnf_release.go
package update

import (
	"fmt"
	"log"
	"os"
	"os/exec"
	"strings"
)

// DNF Release
func DNFRelease() {
	var dnfRelease *exec.Cmd = exec.Command(
		"sudo",
		"dnf",
		"system-upgrade",
		"download",
		"--releasever=41",
	)

	var dnfReleaseOutput strings.Builder

	dnfRelease.Stdout = &dnfReleaseOutput

	var dnfReleaseError error = dnfRelease.Run()

	if dnfReleaseError != nil {
		log.Fatalln("DNF Release Download Failed: \n", dnfReleaseError)
	} else {
		fmt.Println("Executed Command: \n", dnfRelease)
		fmt.Println("DNF Release Download Output: \n", dnfReleaseOutput.String())

		os.Exit(0)
	}
}

File: upt/src/upgrades/dnf_release.rs
use std::{
    io::{Error, StdoutLock, Write, stdout},
    process::{Command, Output, exit},
    result::{
        Result,
        Result::{Err, Ok},
    },
};

// DNF System Release Upgrade
pub fn release_dnf() -> () {
    let dnf_release: Result<Output, Error> = Command::new("dnf")
        .arg("system-upgrade")
        .arg("download")
        .arg("--releasever=41")
        .output();
    let mut standard_output: StdoutLock = stdout().lock();

    match dnf_release {
        Ok(release) => {
            standard_output.write_all(&release.stdout).unwrap();
            writeln!(standard_output, "{}", release.status).unwrap();
        }
        Err(error) => {
            eprintln!("Error Executing DNF System Release Upgrade: {}", error);
            exit(1);
        }
    };

    return ();
}
